### ISSUE TITLE
Reason for RedisTimeoutExceptions - CPU and Network seems fine?

### ISSUE THREAD

**Issue by thomaswr** [StackExchange/StackExchange.Redis#978]
I'm developing an ASP.NET Core 2.1 web application retrieving Redis values on each request.
But I can't get it to saturate CPU fully and I'm seeing TimeoutExceptions like this one:

```
System.AggregateException: One or more errors occurred. (Timeout awaiting response (5004ms elapsed, timeout is 5000ms),
    inst: 0, qs: 95, in: 475,
    serverEndpoint: ,
    mgr: 10 of 10 available,
    IOCP: (Busy=0,Free=1000,Min=2,Max=1000),
    WORKER: (Busy=13,Free=32754,Min=2,Max=32767),
    v: 2.0.513.63329
    (Please take a look at this article for some common client-side issues that can cause timeouts: https://stackexchange.github.io/StackExchange.Redis/Timeouts))

   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.WaitAllCore(Task[] tasks, Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at StackExchange.Redis.ConnectionMultiplexer.WaitAll(Task[] tasks) in C:\\projects\\stackexchange-redis\\src\\StackExchange.Redis\\ConnectionMultiplexer.cs:line 614
   ...
   
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.Invoke(HttpContext context)
---> (Inner Exception #0) StackExchange.Redis.RedisTimeoutException: Timeout awaiting response (5004ms elapsed, timeout is 5000ms), inst: 0, qs: 95, in: 475, serverEndpoint: , mgr: 10 of 10 available, IOCP: (Busy=0,Free=1000,Min=2,Max=1000), WORKER: (Busy=13,Free=32754,Min=2,Max=32767), v: 2.0.513.63329 (Please take a look at this article for some common client-side issues that can cause timeouts: https://stackexchange.github.io/StackExchange.Redis/Timeouts)<---
---> (Inner Exception #1) StackExchange.Redis.RedisTimeoutException: Timeout awaiting response (5004ms elapsed, timeout is 5000ms), inst: 0, qs: 95, in: 475, serverEndpoint: , mgr: 10 of 10 available, IOCP: (Busy=0,Free=1000,Min=2,Max=1000), WORKER: (Busy=13,Free=32754,Min=2,Max=32767), v: 2.0.513.63329 (Please take a look at this article for some common client-side issues that can cause timeouts: https://stackexchange.github.io/StackExchange.Redis/Timeouts)<---
---> (Inner Exception #2) StackExchange.Redis.RedisTimeoutException: Timeout awaiting response (5004ms elapsed, timeout is 5000ms), inst: 0, qs: 95, in: 475, serverEndpoint: , mgr: 10 of 10 available, IOCP: (Busy=0,Free=1000,Min=2,Max=1000), WORKER: (Busy=13,Free=32754,Min=2,Max=32767), v: 2.0.513.63329 (Please take a look at this article for some common client-side issues that can cause timeouts: https://stackexchange.github.io/StackExchange.Redis/Timeouts)<---
---> (Inner Exception #3) StackExchange.Redis.RedisTimeoutException: Timeout awaiting response (5004ms elapsed, timeout is 5000ms), inst: 0, qs: 95, in: 475, serverEndpoint: , mgr: 10 of 10 available, IOCP: (Busy=0,Free=1000,Min=2,Max=1000), WORKER: (Busy=13,Free=32754,Min=2,Max=32767), v: 2.0.513.63329 (Please take a look at this article for some common client-side issues that can cause timeouts: https://stackexchange.github.io/StackExchange.Redis/Timeouts)<---
---> (Inner Exception #4) StackExchange.Redis.RedisTimeoutException: Timeout awaiting response (5004ms elapsed, timeout is 5000ms), inst: 0, qs: 95, in: 475, serverEndpoint: , mgr: 10 of 10 available, IOCP: (Busy=0,Free=1000,Min=2,Max=1000), WORKER: (Busy=13,Free=32754,Min=2,Max=32767), v: 2.0.513.63329 (Please take a look at this article for some common client-side issues that can cause timeouts: https://stackexchange.github.io/StackExchange.Redis/Timeouts)<---
```

The web application is running on a cloud instance with 2 virtual cores but CPU stays at around 30% when it starts to throw these.
CPU and Network IO on the Redis server seems fine, nowhere near critical.

I read https://stackexchange.github.io/StackExchange.Redis/Timeouts but I can't really figure out which limit I'm hitting here? Any ideas? Thank you!

**Comment by mrantonwinter** [comment]
I am seeing similar issues with .net core 2.1

I believe microsoft is suggesting not to use the default library  Microsoft.AspNetCore.All, but rather 
Microsoft.AspNetCore.App.   

Microsoft.AspNetCore.All has an older version of the stackexchange redis library in it.

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/metapackage?view=aspnetcore-2.1

**Comment by mgravell** [comment]
The combined improvements in 2.0.588 and 2.0.571 should, we believe, address a wide range of timeout and performance problems. I'm closing all timeout issues as "believed fixed in 2.0.588"; if you still see problems, please report them as new issues so we can start investigation with fresh eyes.
